interface/src/project/SettingsCustomization.tsx

```diff
@@ -507,61 +503,59 @@ const SettingsCustomization: FC = () => {
-                              isSet={(de.m & DeviceEntityMask.DV_FAVORITE) === DeviceEntityMask.DV_FAVORITE}
-                            />
-                          </ToggleButton>
-                          <ToggleButton value="4" disabled={!de.w || (de.m & 0x83) >= 3}>
-                            <OptionIcon
-                              type="readonly"
-                              isSet={(de.m & DeviceEntityMask.DV_READONLY) === DeviceEntityMask.DV_READONLY}
-                            />
-                          </ToggleButton>
-                          <ToggleButton value="2" disabled={de.n === '' || (de.m & 0x80) !== 0}>
-                            <OptionIcon
-                              type="api_mqtt_exclude"
-                              isSet={
-                                (de.m & DeviceEntityMask.DV_API_MQTT_EXCLUDE) === DeviceEntityMask.DV_API_MQTT_EXCLUDE
-                              }
-                            />
-                          </ToggleButton>
-                          <ToggleButton value="1" disabled={de.n === undefined || (de.m & 0x80) !== 0}>
-                            <OptionIcon
-                              type="web_exclude"
-                              isSet={(de.m & DeviceEntityMask.DV_WEB_EXCLUDE) === DeviceEntityMask.DV_WEB_EXCLUDE}
-                            />
-                          </ToggleButton>
-                          <ToggleButton value="128">
-                            <OptionIcon
-                              type="deleted"
-                              isSet={(de.m & DeviceEntityMask.DV_DELETED) === DeviceEntityMask.DV_DELETED}
-                            />
-                          </ToggleButton>
-                        </ToggleButtonGroup>
-                      )}
+                          />
+                        </ToggleButton>
+                        <ToggleButton value="1" disabled={de.n === undefined || (de.m & 0x80) !== 0}>
+                          <OptionIcon
+                            type="web_exclude"
+                            isSet={(de.m & DeviceEntityMask.DV_WEB_EXCLUDE) === DeviceEntityMask.DV_WEB_EXCLUDE}
+                          />
+                        </ToggleButton>
+                        <ToggleButton value="128">
+                          <OptionIcon
+                            type="deleted"
+                            isSet={(de.m & DeviceEntityMask.DV_DELETED) === DeviceEntityMask.DV_DELETED}
+                          />
+                        </ToggleButton>
+                      </ToggleButtonGroup>
                     </Cell>
-                    <Cell>{!deviceEntity && formatName(de)}</Cell>
-                    <Cell>{!deviceEntity && !(de.m & DeviceEntityMask.DV_READONLY) && formatValue(de.mi)}</Cell>
-                    <Cell>{!deviceEntity && !(de.m & DeviceEntityMask.DV_READONLY) && formatValue(de.ma)}</Cell>
-                    <Cell>{!deviceEntity && formatValue(de.v)}</Cell>
```

Comment:
On my old laptop it takes ~5 seconds for each character input in dialog if the list is rendered in background. With empty list i can just type a new name without waiting.

Reply:
ok, I wondered why that clause was in there. I'll see if I can optimize it

Reply:
Not rendering the table jumps to top of table after editing, the trick was to render the table, but not the content of the cells. Then scolling stays at the same place.

Reply:
i know, but I'd like to code it properly. the biggest problem with React is that the DOM and all its children are re-rendered when a state changes (in this case the deviceEntity). The solution is to use a pattern like https://www.developerway.com/posts/react-re-renders-guide for the Modals where the props and states are separated. Some coding to do....